workflows/diagnostics/fv3net/diagnostics/offline/_input_sensitivity.py

```diff
@@ -0,0 +1,215 @@
+    """
+    for name in quantity_names:
+        if len(dataset[name].dims) > 2:
+            value = dataset[name]
+            raise ValueError(
+                "can only count 1D/2D (sample[, z]) "
+                f"variables, recieved values for {name} with dimensions {value.dims}"
+            )
+    return_dict = {}
+    for name in quantity_names:
+        value = dataset[name]
+        if len(value.dims) == 1 and value.dims[0] == sample_dim_name:
+            return_dict[name] = 1
+        elif value.dims[0] != sample_dim_name:
+            raise ValueError(
+                f"cannot count values for {name} whose first dimension is not the "
+                f"sample dimension ({sample_dim_name}), has dims {value.dims}"
+            )
+        else:
+            return_dict[name] = value.shape[1]
+    return return_dict
+
+
+def _get_variable_indices(
+    data: xr.Dataset, variables: Sequence[str]
+) -> Mapping[str, Tuple[int, int]]:
+    if "time" in data.dims:
+        data = data.isel(time=0).squeeze(drop=True)
+    stacked = data.stack(sample=["tile", "x", "y"])
+    variable_dims = _count_features_2d(
+        variables, stacked.transpose("sample", ...), "sample"
+    )
+    start = 0
+    variable_indices = {}
+    for var, var_dim in variable_dims.items():
+        variable_indices[var] = (start, start + var_dim)
+        start += var_dim
+    return variable_indices
+
+
+def plot_input_sensitivity(
+    model: fv3fit.Predictor, sample: xr.Dataset, output_path: str
+):
+    base_model = model.base_model if isinstance(model, fv3fit.DerivedModel) else model
+
+    try:
+        jacobians = fv3fit.compute_standardized_jacobians(
+            base_model.dict_compatible_model,
+            dataset_to_dict_input(sample),
+            base_model.input_variables,
+        )
+        fig = _plot_jacobians(jacobians)
+        return fig
+
+    except AttributeError:
+        try:
+            input_feature_indices = _get_variable_indices(
+                data=sample, variables=base_model.input_variables
+            )
+            fig = _plot_rf_feature_importance(input_feature_indices, base_model)
+            return fig
+
+        except AttributeError:
+            logger.info(
+                f"Base model is {type(base_model).__name__}, "
+                "which currently has no feature importance or Jacobian "
+                "calculation implemented."
+            )
+            pass
+
+
+def _plot_rf_feature_importance(
+    input_feature_indices: Mapping[str, Tuple[int, int]], wrapped_model: SklearnWrapper,
+):
+
+    # TODO: After the planned removal of the fv3fit.sklearn.EnsembleRegressor
+    # is complete, this for loop is not needed.
+    importances = []
+    for member in wrapped_model.model.regressors:
+        importances.append(member.feature_importances_)
+    mean_importances = np.array(importances).mean(axis=0)
+    std_importances = np.array(importances).std(axis=0)
```

Comment:
Would be nice to decouple this code from the `SklearnWrapper`...for instance it should avoid looking at 
 `SklearnWrapper.model.regressors`. Can you make `mean_importances` and `std_importances` properties of `SklearnWrapper`?

Reply:
See https://en.wikipedia.org/wiki/Law_of_Demeter